src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
```

Comment:
PullProofAI

The `APIResponse.create` function signature can be improved by using TypeScript's optional properties instead of passing `undefined` values for optional parameters.

```diff
-     message?: string,
-     data?: any,
-     error?: any,
+     message: string = '',
+     data: any = null,
+     error: any = null,
```



<!-- This is an auto-generated comment by PullProofAI -->